src/EditorFeatures/Core/SymbolSearch/SymbolSource.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.FindUsages;
+using Microsoft.CodeAnalysis.Editor.GoToBase;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Core.Imaging;
+using Microsoft.VisualStudio.Language.Intellisense.SymbolSearch;
+
+namespace Microsoft.CodeAnalysis.Editor.SymbolSearch
+{
+    class SymbolSource : ISymbolSourceFromLocation
+    {
+        private SymbolSourceProvider _symbolSourceProvider;
+
+        public SymbolSource(SymbolSourceProvider symbolSourceProvider)
+        {
+            _symbolSourceProvider = symbolSourceProvider;
+        }
+
+        public string UniqueId => "Roslyn symbol source";
+
+        public ImageId DisplayIcon => throw new NotImplementedException();
+
+        public string DescriptionText => "Symbols provided by the local language service";
+
+        public string DisplayName => "Local symbol source";
+
+        public async Task<SymbolSearchStatus> FindSymbolsAsync(string navigationKind, VisualStudio.Language.Intellisense.SymbolSearch.Location sourceLocation, IStreamingSymbolSearchSink sink, CancellationToken token)
```

Comment:
what is the threading on this?  will htis only be called on the UI thread?

Reply:
> @AmadeusW do you have an image showing the UI or otherwise end-user experience for this work? Thanks!

I'm afraid I can't publicly share screenshots until the Ignite conference.

One of the goals is to preserve the existing look and feel of the control, including the bold definition header items that appear when user groups by definitions.

The user experience we're designing builds on top of the existing experience, and for this, the commands must be handled in a centralized location - the Editor.

We are also aware that ultimately parts of this code will be removed when Find All References\Go To (aka Symbol Search) moves to LSP. We're making changes here because Roslyn is the easiest project to tinker with and to demonstrate improvements to the Symbol Search.

Reply:
This method won't be called on the UI thread. Should I explicitly call this out?

Reply:
*Definitely*.

Also, if it *shouldn't* ever be called on hte UI thread, we should also assert that that's the case in our code to catch issues if that regresses.

Reply:
I uploaded a screenshot to the main post of this PR